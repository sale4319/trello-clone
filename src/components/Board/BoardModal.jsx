import React, { useEffect, useState } from 'react';

const BoardModal = (props) => {
    const { onSubmit, onClose, title } = props;

    const [boardName, setBoardName] = useState('');
    const [nameValid, setNameValid] = useState(false);

    useEffect(() => {
        if (boardName) {
            setNameValid(true);
        } else {
            setNameValid(false);
        }
    }, [boardName]);

    const handleKeyPress = (event) => {
        if (event.key === 'Enter' && nameValid) {
            onSubmit(boardName);
            onClose();
        }
    };

    return (
        <div className="board-modal-container">
            <div className="board-modal-content">
                <div className="board-modal-title">
                    <div>{title}</div>
                </div>
                <div className="board-modal-body">
                    <input
                        className="board-modal-input"
                        placeholder="Board name"
                        value={boardName}
                        onChange={event => setBoardName(event.target.value)}
                        onKeyPress={handleKeyPress}
                    />
                </div>

                <div className="board-modal-footer">
                    <div
                        className={
                            `board-modal-button
                            ${nameValid ? 'board-modal-button-confirm-valid' : 'board-modal-button-confirm-invalid'}`
                        }
                        onClick={() => {
                            if (nameValid) {
                                onSubmit(boardName);
                                onClose();
                            }
                        }}>
                        Confirm
                    </div>
                    <div className="board-modal-button board-modal-button-close" onClick={onClose}>
                        Close
                    </div>
                </div>
            </div>
        </div>
    );
};

export default BoardModal;