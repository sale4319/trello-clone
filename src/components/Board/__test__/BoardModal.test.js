import React from 'react';
import BoardModal from '../BoardModal';

import '@testing-library/jest-dom/extend-expect';

import renderer from 'react-test-renderer';

test('component renders', () => {
    renderer.create(<BoardModal />).getInstance();

  });

test("mateches snapshot", () => {
    const tree = renderer.create(<BoardModal/>).toJSON();
    expect(tree).toMatchSnapshot();
});