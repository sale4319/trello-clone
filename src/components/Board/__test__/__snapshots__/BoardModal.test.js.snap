// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mateches snapshot 1`] = `
<div
  className="board-modal-container"
>
  <div
    className="board-modal-content"
  >
    <div
      className="board-modal-title"
    >
      <div />
    </div>
    <div
      className="board-modal-body"
    >
      <input
        className="board-modal-input"
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="Board name"
        value=""
      />
    </div>
    <div
      className="board-modal-footer"
    >
      <div
        className="board-modal-button
                            board-modal-button-confirm-invalid"
        onClick={[Function]}
      >
        Confirm
      </div>
      <div
        className="board-modal-button board-modal-button-close"
      >
        Close
      </div>
    </div>
  </div>
</div>
`;
